index.bs

```diff
@@ -232,27 +232,137 @@ Non-Functional Requirements {#requirements-non-functional}
     APIs.
  
 
-Receiver Discovery {#discovery}
+Discovery with mDNS {#discovery}
 ===============================
 
-Issue(100): Incorporate material from the [mDNS](mdns.md) document.
-
-Transport Establishment {#transport}
-====================================
-
-Issue(101): Incorporate material from the \[QUIC](quic.md) document.
+Endpoints may discover one another using
+[DNS-SD](https://tools.ietf.org/html/rfc6763) over
+[mDNS](https://tools.ietf.org/html/rfc6762).  To do so, endpoints must
+use the service name "_openscreen._udp.local".
+
+Advertising endpoints must include DNS TXT records with the following
+keys and values:
+
+- key "fp" with value of the certificate fingerpint of the advertising endpoint.
+  The format of the fingerprint is defined by [RFC 8122 section
+  5](https://tools.ietf.org/html/rfc8122#section-5), excluding the
+  "fingerprint:" prefix and including the hash function, space, and hex-encoded
+  fingerprint.  The fingerprint value also functions as an ID for the endpoint.
+  All endpoints must support the following hash functions: "sha-256", "sha-512".
+  Endpoints must not support the following hash functions: "md2", "md5". 
+
+- key "time" with arbitrary unsigned integer value.  To indicate that
+  metadata has changed, the advertising endpoint must update it to a
+  greater value.  This signals to the discovering endpoint that it
+  should connect to the advertising endpoint to discover updated
+  metadata.
+ 
+An endpoint that advertises this service name must implement the
+transport and metadata discovery mechanism with QUIC described below.
+If a future version of this protocol uses mDNS but breaks
+compatibility with that mechanism, it should change the service name
+to a new value indicating a new mechanism for metadata discovery
+mechanism.  If future versions of this protocol or extensions to this
+protocol are compatible with the metadata discovery mechanism, they
+can use it to indicate support for future versions and extensions.
```

Comment:
No, what I meant was if we have new things added (like streaming), as long as we continue to use QUIC and CBOR to learn that one does the new thing, one does not need to change the service name.  But if one change away from QUIC or CBOR, one would need to change the service name because QUIC and CBOR are the things used to learn about new things.

It's a bit tricky to explain.  Ideas for how to improve are welcome.

Reply:
My suggestion for the last two sentences:

> Future extensions to this QUIC-based protocol can use the same metadata discovery process to indicate support for those extensions, through a capabilities mechanism to be determined.  If a future version of the Open Screen Protocol uses mDNS but breaks compatibility with the metadata discovery process, it should change the DNS-SD service name to a new value, indicating a new mechanism for metadata discovery.

Reply:
I like it.  I replaced the current text with that text.